plenty_api/api.py

```diff
@@ -518,43 +524,85 @@ def __repeat_get_request_for_all_unpaginated_records(self,
-                                                 path=path,
-                                                 query=query)
-            if not response:
-                return None
+        if len(response[page_info['data']]) == query['itemsPerPage']:
+            while True:
+                logging.debug(f"fetching page {query['page'] + 1}")
+                query.update({'page': query['page'] + 1})
+                response = self.__plenty_api_request(method='get',
+                                                    domain=domain,
+                                                    path=path,
+                                                    query=query)
+                if not response:
+                    return None
 
-            if isinstance(response, dict) and 'error' in response.keys():
-                logging.error(f"subsequent {domain} API requests failed.")
-                return response
+                if isinstance(response, dict) and 'error' in response.keys():
+                    logging.error(f"subsequent {domain} API requests failed.")
+                    return response
 
-            entries += response[page_info['data']]
+                entries += response[page_info['data']]
 
-            if len(response[page_info['data']]) < query['itemsPerPage']:
-                break
+                if len(response[page_info['data']]) < query['itemsPerPage']:
+                    break
 
         return entries
 
 
+    def __get_request_dump_file(self,
+                                             domain: str,
+                                             query: dict,
+                                             path: str = '',
+                                             download_filepath: str = '') -> dict:
+        """
+        Dumping raw response to file.
+        Used to download files via Plenty REST API
 
+        Parameter:
+            domain      [str]   -   bi_raw...
+            query       [dict]  -   Additional options for the request
 
+        Return:
+                        [dict]  -   filename on success or response on error
+        """
+        response = self.__plenty_api_request(method='get',
+                                             domain=domain,
+                                             path=path,
+                                             query=query)
+        
+        
+        
+        if not response:
+            return None
 
-    def plenty_api_get_bi_raw_files(self, refine: dict) -> list:
+        if ((isinstance(response, dict) and 'error' in response.keys()) or
+                isinstance(response, list)):
+            return response
+        
+        try:
+            requested_file = open(download_filepath, "wb")
+        except IOError as e:
+            raise
+        else:
+            with requested_file:
+                requested_file.write(response)
+                return {'file': download_filepath}
```

Comment:
Here you catch the error and then you directly raise again, which has no upsides.
Plus, you can combine the open and the with lines:
```python
-        try:
-            requested_file = open(download_filepath, "wb")
-        except IOError as e:
-            raise
-        else:
-            with requested_file:
-                requested_file.write(response)
-                return {'file': download_filepath}
 
+        with open(download_filepath, "wb") as requested_file:
+            requested_file.write(response)
+            return {'file': download_filepath}
```